pkg/api/v1/field_patterns.go

```diff
@@ -0,0 +1,44 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package v1
+
+import (
+	"github.com/go-openapi/errors"
+	"github.com/go-openapi/validate"
+)
+
+// NO TESTS
+
+// FieldPattern field pattern
+// swagger:model FieldPattern
+type FieldPattern struct {
+
+	// regex pattern
+	Pattern string `json:"pattern"`
+
+	// translated msg presented to user
+	Message string `json:"message"`
+}
+
+// FieldPatternWordNumberDashOnly letter, underscore, number and dash
+var FieldPatternWordNumberDashOnly = FieldPattern{
+	Pattern: `^[\w\d][\w\d\-]*$`,
+	Message: " should start with letter or number and may only contain letters, numbers, underscores and dashes",
+}
+
+// FieldPatternLetterNumberDashOnly letter, number and dash
+var FieldPatternLetterNumberDashOnly = FieldPattern{
```

Comment:
Do we ever use this?

Reply:
For now, this one was only used in EventDriver name.